File: src/runtime/php/std/time_native.php
<?php
declare(strict_types=1);
/**
 * Native implementation of pytra.std.time for PHP.
 *
 * Provides __native_<func> implementations that the emitter-generated
 * std/time.php delegates to via @extern convention.
 *
 * source: src/pytra/std/time.py
 */

if (!function_exists('__native_perf_counter')) {
    function __native_perf_counter(): float {
        return microtime(true);
    }
}

File: src/runtime/go/std/time_native.go
// time_native.go: @extern delegation for pytra.std.time.
// Hand-written native implementation.
package main

import "time"

var _pytra_perf_counter_start = time.Now()

func perf_counter() float64 {
	return float64(time.Since(_pytra_perf_counter_start).Nanoseconds()) / 1e9
}
func py_perf_counter() float64 { return perf_counter() }
